dotnet/src/SemanticKernel/SkillDefinition/SKFunction.cs

```diff
@@ -1033,5 +1040,29 @@ private static void TrackUniqueParameterType(ref bool hasParameterType, MethodIn
     /// <summary>Formatter functions for converting parameter types to strings.</summary>
     private static readonly ConcurrentDictionary<Type, Func<object?, CultureInfo, string>?> s_formatters = new();
 
+    /// <summary>Add default values to the context variables if the variable is not defined</summary>
+    private List<string> AddDefaultValues(ContextVariables variables)
+    {
+        var addedVariables = new List<string>();
+        foreach (var parameter in this.Parameters)
+        {
+            if (!variables.ContainsKey(parameter.Name) && parameter.DefaultValue != null)
```

Comment:
Should we also handle the case when parameter exists in context variables, but it's null or empty?

Reply:
The current thinking is only use defaults if the parameter has not been set. Depending on the prompt null or empty string could be expected values.

@shawncal what do you think?

Reply:
Yeah, I'd prefer to use default value if `null` (not set). If user has set value to `string.Empty`, that's a valid, explicit choice that could be a way of saying "ignore this value -- add nothing to the prompt here" and a way of inserting optional clauses into a prompt.